prefect_dask/exceptions.py

```diff
@@ -0,0 +1,9 @@
+"""
+Exceptions specific to prefect-dask.
+"""
+
+
+class ImproperClientError(Exception):
```

Comment:
This name is a little awkward. I've having a hard time coming up with a better suggestion though.

> Raised when the flow/task is async but the client is sync.

Why can't you just let them get a sync client here? It's not good practice to use sync clients in an async context but it should work still, right?

Reply:
Yes, it should still work, but if I name it `get_dask_client` I think this should raise an error.

Reply:
I'm don't think I agree. This seems solvable by documentation in `get_dask_client`. A responsible user of async should notice that they're making calls without `await` and look for the proper context manager. For example, in the Python standard library there are functions that are synchronous and asynchronous. The synchronous ones have a normal name e.g. `contextmanager` and when you need an asynchronous one you look for it...  e.g. `asyncontextmanager`. It's not our responsibility to enforce that you're using the correct one. If a user has a synchronous function that you're calling from your asynchronous task/flow that then uses this client, they will _need_ the synchronous client but this implementation would prevent them from acquiring it.